txn_writer.go

```diff
@@ -288,17 +284,12 @@ func (payload SafeWriterPayload) validate(root string, sm gps.SourceManager) err
 // This mostly guarantees that dep cannot exit with a partial write that would
 // leave an undefined state on disk.
 func (sw *SafeWriter) Write(root string, sm gps.SourceManager, noExamples bool) error {
-
-	if sw.Payload == nil {
-		return errors.New("Cannot call SafeWriter.Write before SafeWriter.Prepare")
-	}
```

Comment:
In this case, we don't have to care about it - that check is an internal integrity check. But "integrity" was only a thing when there was a payload struct that _could_ be uninitialized. Remove the payload, and the necessity of the check goes away.

This check, below:

```if !sw.HasManifest() && !sw.HasLock() && !sw.HasVendor() {```

defines the behavior in the zero value case - which is, "do nothing".

More generally: programatically, we can't prevent people from writing that code, and it is folly to try 😁 . This is the flipside of the Go convention to ["have a useful zero value"](https://golang.org/doc/effective_go.html#allocation_new). If you don't have a useful zero value, then developers can footgun. If it can be unambiguously determined that the programmer made an error (e.g. `new(SafeWriter)`, then a panic is probably appropriate.